test/grep.js

```diff
@@ -127,6 +134,15 @@ test('one file, * in string-regex, make sure * is not globbed', t => {
   t.is(result.toString(), 'this line ends in.js\nlllllllllllllllll.js\n');
 });
 
+test("one file, pattern doesn't match", t => {
```

Comment:
Do you want this under 'valids' or 'invalids?
Even though it's a non-0 exit code,  I would personally consider testing for the absence of something to be a "valid" use case, but I'm happy to move it.

Reply:
This spot LGTM